### ISSUE TITLE
Handling Unicode 

### ISSUE THREAD

**Issue by ssteinbach** [markreidvfx/pyaaf2#33]
Hi Mark,

Hit some more unicode snags from our AAF adapter:

```
advanced_authoring_format.py", line 56, in _walk_item
    slot = thing.slot
  File "/home/steinbach/sw/src/pyaaf2/aaf2/components.py", line 145, in slot
    slot_id = self.slot_id
  File "/home/steinbach/sw/src/pyaaf2/aaf2/components.py", line 126, in slot_id
    return self['SourceMobSlotID'].value
  File "/home/steinbach/sw/src/pyaaf2/aaf2/properties.py", line 145, in value
    return self.typedef.decode(d)
  File "/home/steinbach/sw/src/pyaaf2/aaf2/types.py", line 121, in decode
    assert len(data) == self.size
  File "/home/steinbach/sw/src/pyaaf2/aaf2/types.py", line 95, in size
    return unpack('B', data)[0]
```

And:
```
advanced_authoring_format.py", line 220, in _transcribe
    metadata["Length"] = item.length    
  File "/home/steinbach/sw/src/pyaaf2/aaf2/components.py", line 23, in length                                                                           
    return self['Length'].value                                     
  File "/home/steinbach/sw/src/pyaaf2/aaf2/properties.py", line 145, in value                                                           
    return self.typedef.decode(d)                                     
  File "/home/steinbach/sw/src/pyaaf2/aaf2/types.py", line 704, in decode                                                               
    return self.renamed_typedef.decode(data)
  File "/home/steinbach/sw/src/pyaaf2/aaf2/types.py", line 121, in decode                                                                               
    assert len(data) == self.size                                   
  File "/home/steinbach/sw/src/pyaaf2/aaf2/types.py", line 95, in size                                                                  
    return unpack('B', data)[0]         
TypeError: Struct() argument 1 must be string, not unicode ```

Any suggestions?

**Comment by markreidvfx** [comment]
Before python 2.7.7 the struct module didn't support unicode fmt strings.
https://bugs.python.org/issue19099

This is a issue that the other stefan has identified in other places
before.

doing this should fix it
unpack(str('B'), data)

its a easy thing to forget to do, will fix

**Comment by ssteinbach** [comment]
Awesome, thanks Mark.

**Comment by markreidvfx** [comment]
using `unpack(b'B', data)` also fixes it, not sure if that causes issues in very early versions of python 3, but I prefer doing it that way. Issue was also seen before in #13

should be fixed in 15c8a626aed456fb5b562c8200c20fabb4cedf3a
